yt/visualization/plot_window.py

```diff
@@ -953,19 +953,20 @@ def _setup_plots(self):
                     )
                 elif not np.any(np.isfinite(image)):
                     msg = f"Plot image for field {f} is filled with NaNs."
-                elif np.nanmax(image) > 0.0 and np.nanmin(image) < 0:
+                elif np.nanmax(image) > 0.0 and np.nanmin(image) <= 0:
                     msg = (
                         "Plot image for field %s has both positive "
-                        "and negative values. Min = %f, Max = %f."
+                        "and negative/zero values. Min = %f, Max = %f."
                         % (f, np.nanmin(image), np.nanmax(image))
                     )
                     use_symlog = True
```

Comment:
Now that I understand the underlying problem here...it may be better to keep existing symlog check at line 956 but then add a new check to switch to symlog if there is a zero value and if the data range spans a large order of magnitude. e.g., this seems to work ok: 

```python
                elif np.nanmax(image) > 0.0 and np.nanmin(image) == 0:
                    # normally, a LogNorm scaling would still be OK here because
                    # LogNorm will mask 0 values when calculating vmin. But
                    # due to a bug in matplotlib's imshow, if the data range
                    # spans many orders of magnitude while containing zero points
                    # vmin can get rescaled to 0, resulting in an error when the image
                    # gets drawn. So here we switch to symlog to avoid that until
                    # a fix is in -- see PR #3161 and linked issue.  
                    cutoff_sigdigs = 15   
                    if np.log10(np.nanmax(image)) - np.log10(np.nanmin(image[image>0])) > cutoff_sigdigs:
                        msg = (
                            "Plot image for field %s has both positive "
                            "and zero values with large dynamic range."
                        )
                        use_symlog = True
```

This should pass the image comparisons. The somewhat arbitrary `cutoff_sigdigs` isn't ideal... but maybe there's a smarter way to do this?